internals/garbage-collector.rst

```diff
@@ -319,10 +319,15 @@ the same object survives another GC round in this new generation (generation 1)
 it will be moved to the last generation (generation 2) where it will be
 surveyed the least often.
 
-Generations are collected when the number of objects that they contain reaches some
-predefined threshold, which is unique for each generation and is lower the older
-the generations are. These thresholds can be examined using the  ``gc.get_threshold``
-function:
+In order to decide when to run, the collector keeps track of the number object
+allocations and deallocations since the last collection. When the number of
+allocations minus the number of deallocations exceeds ``threshold_0``,
+collection starts. Initially only generation 0 is examined. If generation 0 has
+been examined more than ``threshold_`` times since generation 1 has been
+examined, then generation 1 is examined as well. With the third generation,
+things are a bit more complicated, see `Collecting the oldest generation`_ for
```

Comment:
```suggestion
things are a bit more complicated, see :ref:`gc-oldest-generation` for
```

Per best practice and standard docs/devguide convention, `:ref:` is used for referencing sections instead of old-style implicit hyperlinks.

You'll also need to add a ref target label before the section heading:

```rst
.. _gc-oldest-generation:

Collecting the oldest generation
--------------------------------
```

Reply:
I am afraid I am not familiar with this practice, what's the rationale for this?

Reply:
This is the standard way to reference sections in Sphinx and the CPython docs See the [quickref](https://devguide.python.org/documentation/markup/#quick-reference) and the [internal links section](https://devguide.python.org/documentation/markup/#internal-links) of the reST Markup page in the devguide itself, and the [Sphinx docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-ref) for full details.

Unlike legacy docutils-style implicit internal hyperlinks, they work reliably instead of silently breaking if section titles are changed at all, either the source or the target is moved to another page, or two sections have the same title, for instance—and in case they do ever break or are incorrect, they will raise a warning on build and appear unlinked in the rendered text instead of failing silently for both. They also automatically insert and update the section title, and they work consistently across all builders. Finally, they are standard convention in the CPython docs and particularly the devguide: here, `:ref:` is used 216 times, compared to only 6 (in 4 in 1 file, plus 1 one-offs) implicit section hyperlinks.

Reply:
Great, thanks for the explanation :+1:

Reply:
Of course, glad it was helpful!